modules/search/instant-search/components/api.js

```diff
@@ -3,26 +3,73 @@
  */
 import fetch from 'unfetch';
 
-const FIELDS = [
-	'author',
-	'comment_count',
-	'date',
-	'excerpt_html',
-	'gravatar_url',
-	'permalink.url.raw',
-	'title_html',
-];
+/**
+ * @preserve jquery-param (c) 2015 KNOWLEDGECODE | MIT
+ * @param {object} a - parameters to add
+ * @returns {string} url query string
+ * From https://github.com/knowledgecode/jquery-param
+ */
+function query_params( a ) {
+	//eslint-disable-next-line no-var
+	var s = [];
+	const add = function( k, v ) {
+		v = typeof v === 'function' ? v() : v;
+		//eslint-disable-next-line no-nested-ternary
+		v = v === null ? '' : v === undefined ? '' : v;
+		s[ s.length ] = encodeURIComponent( k ) + '=' + encodeURIComponent( v );
+	};
+	const buildParams = function( prefix, obj ) {
```

Comment:
I think it'd be nice to add some comments here (and all over, in general) with some examples of the params and what we're doing with them, to make things easier to parse all in all. Anything that can help one jump in and quickly skim functions to understand what we're doing.